sdk/azcore/arm/internal/resource/resource_identifier.go

```diff
@@ -123,9 +123,9 @@ func newResourceIDWithProvider(parent *ResourceID, providerNamespace, resourceTy
 }
 
 func chooseResourceType(resourceTypeName string, parent *ResourceID) ResourceType {
-	if strings.EqualFold(resourceTypeName, resourceGroupsLowerKey) && parent != nil && parent.ResourceType.String() == SubscriptionResourceType.String() {
+	if strings.EqualFold(resourceTypeName, resourceGroupsLowerKey) && isSubscriptionResource(parent) {
```

Comment:
The impl of `isSubscriptionResource()` has a slightly different condition that what's replaced (the addition of `isTenantResource(resourceID.Parent)`). Is this intentional?

Comment applies to other places that use the new helpers.

Reply:
Yes. I fixed all the place that specialize handle the subscription resource and resource group resource, to prevent some resource with pattern of `resourceGroups` and `subscriptions`, but is not a real resource group resource or subscription resource.